meta2v2/meta2_utils.c

```diff
@@ -940,6 +927,57 @@ _real_delete_and_save_deleted_beans(struct sqlx_sqlite3_s *sq3, GSList *beans,
 	return err;
 }
 
+static GError*
+_real_delete_aliases(struct sqlx_sqlite3_s *sq3, GPtrArray *aliases,
+		m2_onbean_cb cb, gpointer u0) {
+	if (aliases->len == 0)
+		return NULL;
+
+	GError *err = NULL;
+	GPtrArray *tmp = g_ptr_array_new();
+	g_ptr_array_add(tmp, NULL);
+	struct bean_ALIASES_s *alias = NULL;
+	struct bean_CONTENTS_HEADERS_s *header = NULL;
+	GSList *beans = NULL;
+	GSList *deleted_beans = NULL;
+
+	void _search_alias_and_header(gpointer plist, gpointer bean) {
+		if (DESCR(bean) == &descr_struct_ALIASES)
+			alias = _bean_dup(bean);
+		else if (DESCR(bean) == &descr_struct_CONTENTS_HEADERS)
+			header = bean;
+		*((GSList**)plist) = g_slist_prepend (*((GSList**)plist), bean);
+	}
+
+	for (guint i = 0; !err && i < aliases->len; i++) {
+		tmp->pdata[0] = aliases->pdata[i];
+		alias = NULL;
+		header = NULL;
+		beans = NULL;
+
+		err = _alias_fetch_info(sq3, 0, tmp, _search_alias_and_header, &beans);
+		if (!err) {
+			if (cb) {
+				deleted_beans = NULL;
+				err = _real_delete_and_save_deleted_beans(sq3, beans, alias,
+						header, _bean_list_cb, &deleted_beans);
+				cb(u0, deleted_beans);
```

Comment:
We can pass `cb` directly to `_real_delete_and_save_deleted_beans`, don't we?

Reply:
`cb` loads a bean list into a list. So `_real_delete_and_save_deleted_beans` creates this bean list.